File: trunk/src/org/jmol/awtjs/JmolFileAdapter.java
package org.jmol.awtjs;

import java.io.BufferedInputStream;
import java.io.IOException;
import java.net.URL;
import java.net.UnknownServiceException;

import org.jmol.api.JmolFileAdapterInterface;

public class JmolFileAdapter implements JmolFileAdapterInterface {

  public Object getBufferedFileInputStream(String name) {
  	// this could be replaced by JavaScript
    try {
      throw new UnknownServiceException("No local file reading in JavaScript version of Jmol");
    } catch (IOException e) {
      return e.getMessage();
    }
  }

	public Object getBufferedURLInputStream(URL url, byte[] outputBytes,
			String post) {
		try {
			JmolURLConnection conn = (JmolURLConnection) url.openConnection();
			String type = null;
			if (type != null) {
				if (outputBytes == null)
					conn.outputString(post);
				else
					conn.outputBytes(outputBytes);
			}
			return new BufferedInputStream(conn.getInputStream());
		} catch (IOException e) {
			return e.getMessage();
		}
	}


}

File: js/org/jmol/awtjs/JmolFileAdapter.js
Clazz.declarePackage ("org.jmol.awtjs");
Clazz.load (["org.jmol.api.JmolFileAdapterInterface"], "org.jmol.awtjs.JmolFileAdapter", ["java.io.BufferedInputStream", "java.net.UnknownServiceException"], function () {
c$ = Clazz.declareType (org.jmol.awtjs, "JmolFileAdapter", null, org.jmol.api.JmolFileAdapterInterface);
Clazz.overrideMethod (c$, "getBufferedFileInputStream", 
function (name) {
try {
throw  new java.net.UnknownServiceException ("No local file reading in JavaScript version of Jmol");
} catch (e) {
if (Clazz.exceptionOf (e, java.io.IOException)) {
return e.getMessage ();
} else {
throw e;
}
}
}, "~S");
Clazz.overrideMethod (c$, "getBufferedURLInputStream", 
function (url, outputBytes, post) {
try {
var conn = url.openConnection ();
var type = null;
if (type != null) {
if (outputBytes == null) conn.outputString (post);
 else conn.outputBytes (outputBytes);
}return  new java.io.BufferedInputStream (conn.getInputStream ());
} catch (e) {
if (Clazz.exceptionOf (e, java.io.IOException)) {
return e.getMessage ();
} else {
throw e;
}
}
}, "java.net.URL,~A,~S");
});
